code/game/machinery/digital_clock.dm

```diff
@@ -0,0 +1,161 @@
+	result_path = /obj/machinery/digital_clock
+	pixel_shift = 28
+
+/obj/machinery/digital_clock/wrench_act_secondary(mob/living/user, obj/item/tool)
+	. = ..()
+	balloon_alert(user, "[anchored ? "un" : ""]securing...")
+	tool.play_tool_sound(src)
+	if(tool.use_tool(src, user, 6 SECONDS))
+		playsound(loc, 'sound/items/deconstruct.ogg', 50, TRUE)
+		balloon_alert(user, "[anchored ? "un" : ""]secured")
+		deconstruct()
+		return TRUE
+
+/obj/machinery/digital_clock/welder_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return
+	if(atom_integrity >= max_integrity)
+		balloon_alert(user, "it doesn't need repairs!")
+		return TRUE
+	user.balloon_alert_to_viewers("repairing display...", "repairing...")
+	if(!tool.use_tool(src, user, 4 SECONDS, amount = 0, volume=50))
+		return TRUE
+	balloon_alert(user, "repaired")
+	atom_integrity = max_integrity
+	set_machine_stat(machine_stat & ~BROKEN)
+	update_appearance()
+	return TRUE
+
+/obj/machinery/digital_clock/multitool_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return
+	if(!(obj_flags & EMAGGED))
+		return
+	balloon_alert(user, "resetting")
+	tool.play_tool_sound(src)
+	if(tool.use_tool(src, user, 6 SECONDS))
+		playsound(loc, 'sound/items/deconstruct.ogg', 50, TRUE)
+		balloon_alert(user, "reset")
+		obj_flags &= ~EMAGGED
+		return TRUE
+
+/obj/machinery/digital_clock/emag_act(mob/user)
+	if(obj_flags & EMAGGED)
+		return
+	playsound(src, SFX_SPARKS, 100, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	do_sparks(3, FALSE, src)
+	obj_flags |= EMAGGED
+
+/obj/machinery/digital_clock/emp_act(severity)
+	. = ..()
+	emag_act()
+
+/obj/machinery/digital_clock/deconstruct(disassembled = TRUE)
+	if(flags_1 & NODECONSTRUCT_1)
+		return
+	if(!disassembled)
+		new /obj/item/stack/sheet/iron(drop_location(), 2)
+		new /obj/item/shard(drop_location())
+		new /obj/item/shard(drop_location())
+	else
+		new /obj/item/wallframe/digital_clock(drop_location())
+	qdel(src)
+
+//MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/digital_clock, 32)
+
+/obj/machinery/digital_clock/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSdigital_clock, src)
+	update_appearance()
+
+/obj/machinery/digital_clock/Destroy()
+	STOP_PROCESSING(SSdigital_clock, src)
+	. = ..()
+
+/obj/machinery/digital_clock/process(seconds_per_tick)
+	if(machine_stat & NOPOWER)
+		return
+	update_appearance()
+
+
+/obj/machinery/digital_clock/update_appearance(updates=ALL)
+	. = ..()
+	if(machine_stat & (NOPOWER|BROKEN))
+		set_light(0)
+		return
+	set_light(1.5, 0.7, LIGHT_COLOR_BLUE) // blue light
+
+/obj/machinery/digital_clock/update_overlays()
+	. = ..()
+
+	if(machine_stat & (NOPOWER|BROKEN))
+		return
+	. += update_time()
+	return .
+
+/obj/machinery/digital_clock/proc/update_time()
+
+	var/station_minutes
+	if(obj_flags & EMAGGED)
+		station_minutes = rand(0, 99)
+	else
+		station_minutes = text2num(station_time_timestamp(format = "mm"))
+
+	// tence / the '3' in '31' / 31 -> 3.1 -> 3
```

Comment:
Do you mean tenths? I can't find this on Google

Reply:
i saw it on one googl answer but ok i uesss